### PR TITLE
Refactor Game class (part 2)

### PR DESCRIPTION
**Pull request by nerzhul**
* showPauseMenu is now part of game
* remove many flags parameters passed to game functions, use the member.
* rename VolatileRunFlags to GameUIFlags (this will permit to share structure with client and CSM
* updatePointedThing: remove pointer ref, we already have the pointer in rundata
* move some attributes outside of VolatileRunFlags after renaming, to game class
* rename statustext to m_statustext
* make some const variables static
* All those changes permit to reduce a little bit function class cost and will permit to interface CSM with some interesting Game flags

 Expose GameUIFlags to client

* Client now have GameUIFlags parameter and setters for other classes
* [CSM] Fix minimap show/hide in Lua because we now have access to the real flag

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -123,16 +123,15 @@ struct TextDestPlayerInventory : public TextDest {
 	Client *m_client;
 };
 
-struct LocalFormspecHandler : public TextDest {
-	LocalFormspecHandler();
-
-	LocalFormspecHandler(std::string formname) :
+struct LocalFormspecHandler: public TextDest
```
Comment (SmallJoker) [inline src/game.cpp]: `LocalFormspecHandler : public TextDest`

**Comment by nerzhul** [comment]
@SmallJoker fixed

**Comment by juhdanad** [comment]
This pull introduced a compiler warning, since `GameUIFlags` is declared as a struct, but forward-declared as a class.

**Comment by nerzhul** [comment]
@juhdanad exact, thanks for notice

**Pull request merged**